src/sisl/geom/_neighbors/_finder.py

```diff
@@ -3,17 +3,213 @@
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 from __future__ import annotations
 
-from typing import Optional, Sequence, Tuple, Union
+from functools import cached_property
+from typing import Literal, Optional, Sequence, Tuple, Union
 
 import numpy as np
 
-from sisl import Geometry
+from sisl import Geometry, Lattice
 from sisl.typing import AtomsIndex
 from sisl.utils import size_to_elements
 
 from . import _operations
 
 
+class Interactions:
```

Comment:
Is there a reason these don't have an `__iter__` method?

Reply:
Yes, I don't know what `__iter__` should do here :sweat_smile: This is just a base class that is not used directly (for now).